### DISCUSSION TITLE
[FEATURE] Auto Join Channel On Login

### DISCUSSION THREAD

**Discussion by disclosurez** [azerothcore/azerothcore-wotlk#3924] [Ideas]
**Is your feature request related to a problem? Please describe.**
I have tried this module, but it doesn't work;
https://github.com/azerothcore/mod-chat-login

Basically, when players login they need to be auto joined into a "world" channel, this would help so much in small servers, and help then prosper into bigger ones.

**Describe the solution you'd like**
A fix to this module, or another solution to join players into a channel on login.

**Describe alternatives you've considered**
I've tried this code, but doesn't work. Maybe i'm stupid who knows?

```cpp
void OnLogin(Player *player)
{
    if (ChannelMgr* cMgr = ChannelMgr::forTeam(player->GetTeamId()))
    {
        if (Channel* channel = cMgr->GetJoinChannel("world", 5))
            channel->JoinChannel(player->GetSession()->GetPlayer(), "");
        cMgr->LoadChannels();
    }
}
```

**Additional context**
Some examples of other servers doing this:

https://github.com/mynew3/crus-bud/blob/51134ded5224b857a3408819d2cfce3c11d0f3f8/src/server/scripts/Custom/Player/player_channel_pvp.cpp

**Comment by BarbzYHOOL** [comment]
I allowed issues on the original mod repo though

**Comment by wizzymore** [comment]
https://github.com/azerothcore/mod-world-chat
Maybe this will work for you better

**Comment by disclosurez** [comment]
bump

**Comment by BarbzYHOOL** [comment]
mod-world-chat is working fine but it's not as practical as an auto join channel module
